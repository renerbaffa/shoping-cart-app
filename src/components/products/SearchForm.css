.SearchForm-input {
  padding: 6px;
}

.SearchForm-form {
  display: flex;
}

.SearchForm-button {
  background-color: white;
  padding: 0px 18px;
}
