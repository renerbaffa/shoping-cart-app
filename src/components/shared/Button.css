.Button {
  background-color: white;
  padding: 0px 18px;
}
