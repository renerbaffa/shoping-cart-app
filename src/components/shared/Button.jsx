import React from 'react';

/*
  Button component is an adapter.
  Imagine that in the future we move to react-bootstrap or material-ui
  In this case we only need to replase default button by the library button
*/
const Button = ({ children, ...other }) => (
  <button {...other}>
    {children}
  </button>
);

export default Button;
