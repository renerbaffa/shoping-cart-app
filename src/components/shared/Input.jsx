import React from 'react';

/*
  Input component is an adapter.
  Imagine that in the future we move to react-bootstrap or material-ui
  In this case we only need to replase default input by the library input
*/
const Input = (props) => (
  <input {...props} />
);

export default Input;
