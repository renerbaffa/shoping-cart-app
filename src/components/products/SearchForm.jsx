import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Input from '../shared/Input';
import Button from '../shared/Button';

import './SearchForm.css';

class SearchForm extends Component {
  static propTypes = {
    onFilter: PropTypes.func,
  };

  static defaultProps = {
    onFilter: () => {},
  };

  state = {
    searchText: '',
  }

  handleSearchTextChange = event =>
    this.setState({ searchText: event.target.value });

  handleFilter = (event) => {
    event.preventDefault();
    this.props.onFilter(this.state.searchText);
  }

  render() {
    const { searchText } = this.state;

    return (
      <div>
        <form className="SearchForm-form" onSubmit={this.handleFilter}>
          <Input
            className="SearchForm-input"
            onChange={this.handleSearchTextChange}
            placeholder="Search..."
            type="text"
            value={searchText}
          />
          <Button className="SearchForm-button">
            Search
          </Button>
        </form>
      </div>
    );
  }
}

export default SearchForm;
