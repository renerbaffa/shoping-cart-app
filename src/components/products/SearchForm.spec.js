import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SearchForm from './SearchForm';

import Button from '../shared/Button';
import Input from '../shared/Input';

const NEW_TEXT = 'New text';

describe('<SearchForm />', () => {
  let wrapper;
  let onFilter;

  beforeEach(() => {
    onFilter = jest.fn();
    wrapper = shallow(<SearchForm onFilter={onFilter} />);
  });

  it('should render correctly', () => {
    const searchForm = renderer.create(<SearchForm />).toJSON();
    expect(searchForm).toMatchSnapshot();
  });

  it('should uptade state when text changes', () => {
    expect(wrapper.state().searchText).toBe('');
    wrapper.find(Input).props().onChange({ target: { value: NEW_TEXT }});
    expect(wrapper.state().searchText).toBe(NEW_TEXT);
  });

  it('should call onFilter when clicking on search button', () => {
    wrapper.find(Input).props().onChange({ target: { value: NEW_TEXT }});
    expect(onFilter).not.toHaveBeenCalled();
    wrapper.find('form').props().onSubmit({ preventDefault: () =>{} });
    expect(onFilter).toHaveBeenCalledWith(NEW_TEXT);
  });

  it('should reach 100% of coverage', () => {
    SearchForm.defaultProps.onFilter();
  });
});
