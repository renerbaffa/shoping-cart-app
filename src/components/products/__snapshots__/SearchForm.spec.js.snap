// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> should render correctly 1`] = `
<div>
  <form
    className="SearchForm-form"
    onSubmit={[Function]}
  >
    <input
      className="SearchForm-input"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <button
      className="SearchForm-button Button"
    >
      Search
    </button>
  </form>
</div>
`;
